http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/AutopauseMonitor.scala

```diff
@@ -25,84 +24,95 @@ class AutopauseMonitor[F[_]](
   jupyterDAO: JupyterDAO[F],
   publisherQueue: Queue[F, LeoPubsubMessage]
 )(implicit
-  F: Async[F],
-  metrics: OpenTelemetryMetrics[F],
-  logger: StructuredLogger[F],
   dbRef: DbReference[F],
   ec: ExecutionContext
-) {
+) extends BackgroundProcess[F, RuntimeToAutoPause] {
+  override def name: String =
+    "autopause" // autopauseRuntime is more accurate. But keep the current name so that existing metrics won't break
+  override def interval: scala.concurrent.duration.FiniteDuration = config.autoFreezeCheckInterval
 
-  val process: Stream[F, Unit] =
-    (Stream.sleep[F](config.autoFreezeCheckInterval) ++ Stream.eval(
-      autoPauseCheck
-        .handleErrorWith(e => logger.error(e)("Unexpected error occurred during auto-pause monitoring"))
-    )).repeat
+  def filterCriteria(cluster: RuntimeToAutoPause, now: Instant)(implicit
```

Comment:
I think `isCandidate` or `isAutopausable` would make sense for this method name - it acts on one item and returns a boolean